### PR TITLE
Allow deeplinking to a specific Chat message

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers CC @keybase/picnicsquad @keybase/design 

A present for @mmaxim :)  Adds a "Copy a link to this message" option to the message dropdown, and URL handler for going to that conversation and highlighting the message.

Perhaps this should be behind a feature flag while we try it out?  Since it exposes the deeplinking feature to users, and solidifies the URL format (because people will start keeping and sharing the links).

I plumbed this through `selectConversation`, because we won't know the conversationID for the linked message until we get to the bottom of `selectConversation`.

### CODE REVIEW

**Comment by malgorithms** [comment]
cool feature!

File: shared/constants/chat2/meta.tsx [hunk 1]
```diff
@@ -414,6 +414,16 @@ export const getParticipantSuggestions = (state: TypedState, id: Types.Conversat
   return _getParticipantSuggestionsMemoized(participants, teamType)
 }
 
+export const getConversationLabel = (conv: Types.ConversationMeta, includeChannelName: boolean): string => {
```
Comment (jakob223) [inline shared/constants/chat2/meta.tsx]: maybe call the second arg `alwaysIncludeChannelName`?

**Comment by cjb** [comment]
@mikem How about this, 3fc3e21d7e?  I see the sequencing happening properly in the console, though that doesn't guarantee there's no race.

**Comment by cjb** [comment]
@mmaxim I've revived this PR against up to date master now, want to give it a try? I didn't have to alter the code to get it to do what inbox search seems to, which was surprising because I remember it not working properly under some circumstances last time around..

File: shared/actions/chat2/index.tsx [hunk 2]
```diff
@@ -1151,6 +1151,22 @@ function* loadMoreMessages(
       )
     }
 
+    if (
```
Comment (mmaxim) [inline shared/actions/chat2/index.tsx:1154]: Why is this not in the navigateToThread case above?

File: shared/actions/json/chat2.json [hunk 3]
```diff
@@ -38,6 +37,7 @@
       "teamname?": "string",
       "channelname?": "string",
       "conversationIDKey?": "Types.ConversationIDKey", // we only use this when we click on channel mentions. we could maybe change that plumbing but keeping it for now
+      "highlightMessageID?": "number | null",
```
Comment (chrisnojima) [inline shared/actions/json/chat2.json:40]: this stuff should be number or undefind, not null

**Comment by cjb** [comment]
@mmaxim @chrisnojima Thanks! Got it working with a single thread load, ready for another look.

**Comment by mmaxim** [comment]
Did you try on mobile? Unfortunately we do a pretty bad job in centering on mobile, and I wonder if when using this feature it will be extra tough since you won't know what the message is you are jumping to.

**Comment by cjb** [comment]
@mmaxim yep, seems to work perfectly for me on mobile, tested by taking a link to a very old message in a convo that hadn't been opened on that device, and it loaded and highlighted the msg in the center of the convo

**Comment by cjb** [comment]
@chrisnojima did you want to take another look? I think this is ready to go now

**Pull request merged**